assembly/utils.ts

```diff
@@ -74,12 +74,8 @@ export class BalancedTime {
 
 // @ts-ignore: decorator
 @inline
-export function floorDiv(a: i32, b: i32): i32 {
-  return (a >= 0 ? a : a - b + 1) / b;
-}
-
-export function floorDivI64(a: i64, b: i64): i64 {
-  return (a >= 0 ? a : a - b + 1) / b;
+export function floorDiv<T extends number>(a: T, b: T): T {
+  return (a >= 0 ? a : a - b + 1) / b as T;
```

Comment:
thanks @MaxGraey - I originally wrote a generic function but got tripped up by `Type 'number' is not assignable to type 'T'.` - should have realised a simple cast would fix it!

Reply:
It's ok. Btw we already use same approach for `sign` and `ord` helpers. So now you know!=)